File: api-rh/src/main/java/com/example/api_rh/model/Funcionario.java
package com.example.api_rh.model;

import java.util.Date;

public class Funcionario extends Pessoa{
    private String departamento;
    private Date dataAdmissao;
    private Double salario;


    public Funcionario() {
    }

    public Funcionario(String nome, String email, String telefone, String cargo, String departamento, Double salario, String cidade, String status, Date dataAdmissao, Date dataEntrevista) {
        super(nome, email, telefone, cargo, cidade, status);
        this.dataAdmissao = dataAdmissao;
        this.salario = salario;
        this.departamento = departamento;
    }

    public String getDepartamento() {
        return departamento;
    }

    public void setDepartamento(String departamento) {
        this.departamento = departamento;
    }

    public Date getDataAdmissao() {
        return dataAdmissao;
    }

    public void setDataAdmissao(Date dataAdmissao) {
        this.dataAdmissao = dataAdmissao;
    }

    public Double getSalario() {
        return salario;
    }

    public void setSalario(Double salario) {
        this.salario = salario;
    }
}

File: web/src/types/funcionario.ts
export type StatusFuncionario = 'ATIVO' | 'INATIVO' | string;

export interface Funcionario {
  id: number;
  nome: string;
  email: string;
  telefone: string;
  cargo: string;
  departamento: string;
  cidade: string;
  status: StatusFuncionario;
  salario?: number;
  dataAdmissao?: string | Date;
}

export type FuncionarioModalType = 'cadastrar' | 'visualizar' | 'editar' | 'excluir' | null;

export interface FuncionarioFormData {
  id?: number;
  nome: string;
  email: string;
  telefone: string;
  cargo: string;
  departamento: string;
  salario?: number;
  cidade: string;
  status: string;
  dataAdmissao?: string | Date;
}
